objective.py

```diff
@@ -5,70 +5,86 @@
+        for g in range(self.n_groups):
+            grp_g_indices = self.grp_indices[
+                self.grp_ptr[g]: self.grp_ptr[g+1]]
+            w_g = w[grp_g_indices]
 
-        grp_indices, grp_ptr = grp_converter(groups, n_features)
-        n_groups = len(grp_ptr) - 1
-        weights = np.ones(n_groups)
+            sum_penalty += norm(w_g)
+        sum_penalty *= (1 - self.tau)
+        sum_penalty += self.tau * np.sum(np.abs(w))
 
-        lmbd_max = self._compute_lmbd_max(X, y, grp_indices, grp_ptr)
-        self.lmbd = self.reg * lmbd_max
+        return self.lmbd * sum_penalty
+
+    def set_data(self, X, y, groups):
+        self.n_samples, self.n_features = X.shape
+
+        self.grp_indices, self.grp_ptr = grp_converter(groups, self.n_features)
+        self.n_groups = len(self.grp_ptr) - 1
 
         self.X, self.y = X, y
         self.groups = groups
-        self.grp_indices, self.grp_ptr = grp_indices, grp_ptr
 
-        self.datafit = QuadraticGroup(grp_ptr, grp_indices)
-        self.penalty = WeightedGroupL2(
-            self.lmbd, weights, grp_ptr, grp_indices
-        )
+        lmbd_max = self._compute_lmbd_max()
+        self.lmbd = self.reg * lmbd_max
 
-    def evaluate_result(self, w):
-        X, y = self.X, self.y
-        datafit, penalty = self.datafit, self.penalty
+    def evaluate_result(self, beta):
+        diff = self.y - self.X @ beta
+        datafit = .5 * diff @ diff / self.n_samples
+        pen = self.value_penalty(beta)
 
         return dict(
-            value=datafit.value(y, w, X @ w) + penalty.value(w)
+            value=datafit+pen,
         )
 
     def get_objective(self):
         return dict(
             X=self.X, y=self.y, lmbd=self.lmbd, groups=self.groups,
-            grp_indices=self.grp_indices, grp_ptr=self.grp_ptr
+            grp_indices=self.grp_indices, grp_ptr=self.grp_ptr, tau=self.tau
         )
 
-    def _compute_lmbd_max(self, X, y, grp_indices, grp_ptr):
-        lmbd_max = 0.
-        n_samples = X.shape[0]
-        n_groups = len(grp_ptr) - 1
+    def _compute_lmbd_max(self):
+        lmbd_max_l1 = 0.
+        lmbd_max_group = 0.
+        n_groups = len(self.grp_ptr) - 1
 
+        # Compute lambda max for the L1 part
+        lmbd_max_l1 = norm(self.X.T @ self.y, ord=np.inf) / self.n_samples
+
+        # Compute lambda max for the group lasso part
         for g in range(n_groups):
-            grp_g_indices = grp_indices[grp_ptr[g]: grp_ptr[g+1]]
-            lmbd_max = max(
-                lmbd_max,
-                norm(X[:, grp_g_indices].T @ y) / n_samples
+            grp_g_indices = self.grp_indices[
+                self.grp_ptr[g]: self.grp_ptr[g+1]]
+            lmbd_max_group = max(
+                lmbd_max_group,
+                norm(self.X[:, grp_g_indices].T @ self.y) / self.n_samples
             )
 
+        # Combine the two parts using the tau parameter
+        lmbd_max = self.tau * lmbd_max_l1 + (1 - self.tau) * lmbd_max_group
```

Comment:
I don't think this is correct. The dual norm is the epsilon norm, which is harder than this to compute (that's why we needed gsroptim). 
Test at reg=0.99 and reg=1.01 but I don't think the current formula is the good one.
@Badr-MOUFAD

Reply:
Yes indeed, 
as the lambda max is hard to get in this case, I opted for an upper bound which happen to be cheap to get